zb-classifier.js

```diff
@@ -1056,6 +1056,32 @@ class ZigbeeClassifier {
       'parseNumericTenthsAttr',       // parseValueFromAttr
       CONFIG_REPORT_BATTERY
     );
+    if (node.activeEndpoints[genPowerCfgEndpoint].deviceId !=
```

Comment:
I get that a smart plug would not have this property, but is that the only reason for this conditional? If so, are there other classes of devices we should be excluding?

Reply:
I just followed the code above that changed the name from battery to mains.

One alternative is to create the property with visible: false, and then change it to visible: true if/when the battery percentage gets reported. This would mean that the configuration boolean would have to be removed, otherwise there would be too much messing across layers. If we went with this then I'd suggest there are a few other properties that could use the same treatment.

Reply:
why not reporting battery only for nodes with powerSource = POWERSOURCE.BATTERY
https://github.com/WebThingsIO/zigbee-adapter/blob/d19dedf75061f39917371bcf0caae8c24cbd3f61/zb-node.js#L28

Reply:
Thanks @frederic34, I'm looking into this